client/components/pulsing-dot/index.jsx

```diff
@@ -5,9 +5,47 @@
  */
 
 import React from 'react';
+import { number } from 'prop-types';
 import classnames from 'classnames';
 
-export default function PulsingDot( { active } ) {
-	const className = classnames( 'pulsing-dot', { 'is-active': active } );
-	return <div className={ className } />;
+class PulsingDot extends React.Component {
+	constructor( props ) {
+		super( props );
+
+		this.timeout = null;
+		this.state = {
+			show: false,
+		};
+	}
+
+	componentDidMount() {
+		const { delay } = this.props;
+		this.timeout = setTimeout( () => {
+			this.setState( { show: true } );
+		}, delay );
+	}
+
+	componentWillUnmount() {
+		if ( this.timeout ) {
+			clearTimeout( this.timeout );
+		}
+	}
+
+	render() {
+		const { active } = this.props;
+		const { show } = this.state;
+
+		const className = classnames( 'pulsing-dot', { 'is-active': active } );
+		return show ? <div className={ className } /> : null;
```

Comment:
could rejigger this to avoid the classnames call entirely when show is false:
```
if ( ! show ) { return null; }
const className = ...;
return <div ... />
```